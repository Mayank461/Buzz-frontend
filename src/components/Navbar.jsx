import axios from 'axios';
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { API_URL } from '../config';

export default function Navbar() {
  const [user, setUser] = useState({});
  useEffect(() => {
    axios
      .get(`${API_URL}/user/getUser`, { withCredentials: true })
      .then((res) => {
        setUser(res.data[0]);
        console.log(user);
      })
      .catch((err) => console.log(err.message));
  }, []);

  function handleLogout() {
    window.open(`${API_URL}/auth/logout`, '_self');
  }
  return (
    <>
      <nav className="navbar navbar-light bg-light p-0">
        <div className="container-fluid">
          <div>
            <Link to={'/'}>
              <img
                className="logo"
                src="https://mma.prnewswire.com/media/728150/TO_THE_NEW_Logo.jpg?p=facebook"
              />
            </Link>
          </div>
          <div className="d-flex position-relative">
            <Link to='/profile' className='nav-link text-dark'>
            <div className='d-flex'>
            <div className="">
              <img
                src={user.picture_url}
                className="card-img-top small-round-pic  round-img"
                alt="..."
              />
            </div>
            <div className="d-flex align-items-center ms-2">
              {user.firstname + ' ' + user.lastname}
            </div>
            </div>
            </Link>
          
            <div className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2">
              <i className="fa-brands fa-facebook-messenger "></i>
            </div>
            <div className="position-relative d-flex">
              <div className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2 ">
             
                  <i className="fa-solid fa-user"></i>
             
              </div>
              <div className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0">
                1
              </div>
            </div>
            <div
              onClick={handleLogout}
              className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2 pointer"
            >
              <i className="fa fa-sign-out "></i>
            </div>
          </div>
        </div>
      </nav>
    </>
  );
}
