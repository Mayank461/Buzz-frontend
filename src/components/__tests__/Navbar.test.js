import { render, screen } from '@testing-library/react';
import { useState } from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';



test('should render Navbar Panel', () => {
    const user = { firstname: "John", lastname: "Doe", friends: { myFriends: [], mySentRequests: [{}], myFriendRequests: [{}] } };
    render(<BrowserRouter> <Navbar user={user} /> </BrowserRouter>)
    const text = screen.queryByTestId('userInfo').innerHTML;
    expect(text).toBe("John Doe");
})