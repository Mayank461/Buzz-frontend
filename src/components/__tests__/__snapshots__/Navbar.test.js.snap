// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar Panel 1`] = `
<nav
  className="navbar  navbar-light bg-white p-0"
>
  <div
    className="container-fluid"
  >
    <div
      className="col-6 col-md-8 d-flex"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo"
          src="https://mma.prnewswire.com/media/728150/TO_THE_NEW_Logo.jpg?p=facebook"
        />
      </a>
      <div
        className="d-none d-sm-block mx-3 w-100 align-self-center position-relative"
      >
        <input
          autoComplete="off"
          className="caption px-4 py-2 bg-transparent w-100 border-0 border-bottom form-control"
          onChange={[Function]}
          placeholder="Search by Name or Email"
          type="search"
        />
        <div
          className="dropdown-search w-100 position-absolute left-0"
          style={
            Object {
              "zIndex": 1,
            }
          }
        >
          <ul
            className="list-group overflow-auto"
            style={
              Object {
                "maxHeight": "300px",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="col-6 col-md-4 justify-content-end d-flex position-relative align-items-center"
    >
      <a
        className="nav-link text-dark p-0 p-lg-2"
        href="/profile"
        onClick={[Function]}
      >
        <div
          className="d-flex"
        >
          <i
            className="fa-solid fa-user card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"
          />
          <div
            className=" align-items-center ms-2 d-none d-lg-block d-flex align-self-center"
            data-testid="userInfo"
          >
            John Doe
          </div>
        </div>
      </a>
      <div
        className="pointer d-sm-none d-flex align-items-center  ms-2 round-img border rounded-circle icon-bg text-dark p-2 false"
        onClick={[Function]}
      >
        <i
          className="fa-solid fa-search"
        />
      </div>
      <div
        className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2"
      >
        <a
          className="text-dark"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-house"
          />
        </a>
      </div>
      <div
        className="position-relative d-flex "
      >
        <a
          className="d-flex align-items-center  ms-2 round-img border rounded-circle icon-bg text-dark p-2 text-decoration-none"
          href="/chat"
          onClick={[Function]}
        >
          <i
            className="fas fa-comments"
          />
        </a>
        <div
          className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0"
          title="new_message_count"
        >
          0
        </div>
      </div>
      <div
        className="position-relative d-flex "
      >
        <a
          className="d-flex align-items-center  ms-2 round-img border rounded-circle icon-bg text-dark p-2 "
          href="/friends"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-user"
          />
        </a>
        <div
          className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0"
          title="friend_req_count"
        >
          1
        </div>
      </div>
      <div
        className="d-flex align-items-center ms-2 me-3 round-img border rounded-circle icon-bg text-dark p-2 pointer"
        onClick={[Function]}
        title="logout-btn"
      >
        <i
          className="fa fa-sign-out "
        />
      </div>
    </div>
  </div>
</nav>
`;
